raiden/messages/decode.py

```diff
@@ -35,9 +39,11 @@ def lockedtransfersigned_from_message(message: LockedTransferBase) -> LockedTran
     route_states = list()
     for route_metadata in message.metadata.routes:
         try:
-            rs = RouteState(
-                route_metadata.route, address_to_metadata=route_metadata.address_metadata or {}
-            )
+            address_metadata = route_metadata.address_metadata
+            address_metadata = address_metadata or {
+                message.recipient: fetch_address_metadata(message.recipient)
```

Comment:
Wouldn't this mean that the node would fetch its own metadata from the PFS and none of the other addresses' metadata in the route? 
It handles a received message, so the recipient is the node itself. It should rather fetch the addresses from the other participants in this route. BTW, a node can always create its own metadata by `raiden_service.transport.address_metadata`

Reply:
Yes, but that was a mistake. It was supposed to go through all the addresses in the route. I have just fixed it ✅